app/components/ChartSettings/index.js

```diff
@@ -21,14 +22,25 @@ class ChartSettings extends Component {
     return -1 !== this.state.modules.indexOf(name);
   }
 
+  _toggleModuleCallback(expanded, name) {
+    if (expanded) {
+      this.setState({
+        activeSettings: name,
+      });
+    }
+  }
+
   _renderModule(name) {
     if (!this._hasModule(name)) {
       return null;
     }
 
     // Setup props, handling special cases for Metadata and ColorPalette
     const moduleProps = update({}, {
-      defaultExpand: { $set: this._shouldDefaultExpand() },
+      defaultExpand: { $set: this._shouldDefaultExpand(name) },
+      toggleCallback: {
```

Comment:
The typical naming convention for prop callback is onEventName, ie. onClick, onChange, etc..

`onToggle` - prop name
`onModuleToggle` - parent component handler